cpp/src/io/csv/csv_reader.cu

```diff
@@ -771,7 +771,7 @@ gdf_error read_csv(csv_read_arg *args)
 		gdf_column *gdf = (gdf_column *)malloc(sizeof(gdf_column) * 1);
 
 		gdf->size		= raw_csv->num_records;
-		gdf->dtype		= raw_csv->dtypes[col];
+		gdf->dtype		= raw_csv->dtypes[acol];
```

Comment:
I looked at the failure and it's due to whether dtype detection is enabled or not. If dtypes are inferred, then `raw_csv->dtypes` only contains `num_active_cols` elements, otherwise it'll contain `num_actual_cols` elements.

You can either: (1) Add a check here to use either acol or col, depending on which path was used or (2) fill the `raw_csv->dtype` unused elements with dummy/GDF_invalid dtypes.